src/Bicep.Core.UnitTests/TypeSystem/FunctionResolverTests.cs

```diff
@@ -298,7 +299,9 @@ object[] CreateRow(object returnedLiteral, string functionName, params object[]
                 CreateRow(new[] { "pop" }, "intersection", new[] { "fizz", "buzz", "pop" }, new[] { "snap", "crackle", "pop" }),
                 CreateRow(new[] { "fizz", "buzz", "pop" }, "union", new[] { "fizz", "buzz" }, new[] { "pop" }),
                 CreateRow("fizz", "first", new[] { new[] { "fizz", "buzz", "pop" } }),
+                CreateRow(null, "first", new[] { Array.Empty<string>() }),
                 CreateRow("pop", "last", new[] { new[] { "fizz", "buzz", "pop" } }),
+                CreateRow(null, "last", new[] { Array.Empty<string>() }),
```

Comment:
Since `first` and `last` will now be operating on non-literal types, a dedicated test (like the one for `flatten`, but less complicated) would cover more scenarios.